app-server/subprojects/bounded_context/accessibility/application/src/main/kotlin/club/staircrusher/accessibility/application/port/in/AccessibilityImageService.kt

```diff
@@ -0,0 +1,144 @@
+import club.staircrusher.accessibility.domain.model.AccessibilityImage
+import club.staircrusher.place.application.port.`in`.PlaceApplicationService
+import club.staircrusher.stdlib.di.annotation.Component
+import club.staircrusher.stdlib.persistence.TransactionIsolationLevel
+import club.staircrusher.stdlib.persistence.TransactionManager
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.runBlocking
+import mu.KotlinLogging
+import java.io.ByteArrayOutputStream
+import java.nio.file.Files
+import kotlin.io.path.createDirectory
+
+@Component
+class AccessibilityImageService(
+    private val transactionManager: TransactionManager,
+    private val thumbnailGenerator: ThumbnailGenerator,
+    private val fileManagementService: FileManagementService,
+    private val placeApplicationService: PlaceApplicationService,
+    private val placeAccessibilityRepository: PlaceAccessibilityRepository,
+    private val buildingAccessibilityRepository: BuildingAccessibilityRepository,
+) {
+    private val logger = KotlinLogging.logger {}
+
+    fun migrateImageUrlsToImagesIfNeeded(placeId: String) = transactionManager.doInTransaction(isolationLevel = TransactionIsolationLevel.REPEATABLE_READ) {
+        val place = placeApplicationService.findPlace(placeId) ?: return@doInTransaction
+        val placeAccessibility = placeAccessibilityRepository.findByPlaceId(placeId)
+        val buildingAccessibility = buildingAccessibilityRepository.findByBuildingId(place.building.id)
+
+        if (placeAccessibility?.images?.isEmpty() == true && placeAccessibility.imageUrls.isNotEmpty()) {
+            val placeAccessibilityImages = placeAccessibility.imageUrls.map { AccessibilityImage(imageUrl = it, thumbnailUrl = null) }
+            placeAccessibilityRepository.updateImages(placeAccessibility.id, placeAccessibilityImages)
+        }
+
+        if (buildingAccessibility?.entranceImages?.isEmpty() == true && buildingAccessibility.entranceImageUrls.isNotEmpty()) {
+            val buildingEntranceImages = buildingAccessibility.entranceImageUrls.map { AccessibilityImage(imageUrl = it, thumbnailUrl = null) }
+            buildingAccessibilityRepository.updateEntranceImages(buildingAccessibility.id, buildingEntranceImages)
+        }
+
+        if (buildingAccessibility?.elevatorImages?.isEmpty() == true && buildingAccessibility.elevatorImageUrls.isNotEmpty()) {
+            val buildingElevatorImages = buildingAccessibility.elevatorImageUrls.map { AccessibilityImage(imageUrl = it, thumbnailUrl = null) }
+            buildingAccessibilityRepository.updateElevatorImages(buildingAccessibility.id, buildingElevatorImages)
+        }
+    }
+
+    fun generateThumbnailsIfNeeded(placeId: String) {
+        val thumbnailGenerationRequiredImages = getThumbnailGenerationRequiredImages(placeId)
+        val thumbnailUrls = thumbnailGenerationRequiredImages
```

Comment:
```suggestion
        val generatedThumbnailUrls = thumbnailGenerationRequiredImages
```
이런 변수명이면 읽기 더 쉬울 것 같습니닷

Reply:
63f44b5